boards/arm/rcar_h3_salvatorx/rcar_h3_salvatorx_cr7.dts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2022 BayLibre, SAS
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+/dts-v1/;
+#include <arm/renesas/gen3/r8a77951.dtsi>
+#include "rcar_h3_salvatorx_cr7-pinctrl.dtsi"
+
+/ {
+	model = "Renesas h3 Salvator-X board";
+	compatible = "renesas,h3-salvatorx-cr7";
+
+	chosen {
+		zephyr,sram = &sram0;
+		zephyr,console = &scif1;
+		zephyr,shell-uart = &scif1;
+		zephyr,canbus = &can0;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		user_led_0: led_4 {
+			gpios = <&gpio6 11 GPIO_ACTIVE_HIGH>;
+			label = "led4";
+		};
+		user_led_1: led_5 {
+			gpios = <&gpio6 12 GPIO_ACTIVE_HIGH>;
+			label = "led5";
+		};
+		user_led_2: led_6 {
+			gpios = <&gpio6 13 GPIO_ACTIVE_HIGH>;
+			label = "led6";
+		};
+	};
+
+	gpio_keys {
+		compatible = "gpio-keys";
+		user_button_0: sw20 {
+			gpios = <&gpio6 11 GPIO_ACTIVE_LOW>;
```

Comment:
user_led_n, and user_button_n are using the same GPIO line, what happen when you try to run the button demo ?

Reply:
You're right, perhaps that demo won't work properly...
The problem is that on the schematic Renesas used the same GPIO to connect 1 swtich and 1 LED: indeed if you press the switch even with no firmware running on the CR7, the corresponding LED will be switched off. As a consequence, from the firmware point of view, it all depends on how you configure that GPIO line: if it's an output then you'll control the LED; if it's an input then you'll be able to read the switch (of course this holds if you configure the output in push-pull mode, because if you use open-drain one it's even possible to mix the 2 scenarios, but that's not relevant for the buttons demo).
What I can do is to "shift" the alias for the LEDs, something like:
      led0 = &user_led_1;
      led1 = &user_led_2;
      led2 = &user_led_0;

This would resemble the H3ULCB configuration and should work.
Do you think that this would be acceptable?

Reply:
These alias should target hardware agnostic applications, so just pickup a configuration which works in any cases without conflicts. So that no one had to read the dts nor the schematics of this board to make basic usage working. Advanced users will find a way to have the exact configurations they want.

Reply:
Understood, thanks for the explanation. I fixed aliases accordingly and now the button demo behaves the same as if the h3ulcb dts were used